django_rq/queues.py

```diff
@@ -181,21 +185,36 @@ def get_queues(*queue_names, **kwargs):
     All instances must use the same Redis connection.
     """
     from .settings import QUEUES
-    autocommit = kwargs.get('autocommit', None)
-    queue_class = kwargs.get('queue_class', DjangoRQ)
-    if len(queue_names) == 0:
+
+    if len(queue_names) <= 1:
         # Return "default" queue if no queue name is specified
-        return [get_queue(autocommit=autocommit)]
-    if len(queue_names) > 1:
-        queue_params = QUEUES[queue_names[0]]
-        connection_params = filter_connection_params(queue_params)
-        for name in queue_names:
-            if connection_params != filter_connection_params(QUEUES[name]):
-                raise ValueError(
-                    'Queues must have the same redis connection.'
-                    '"{0}" and "{1}" have '
-                    'different connections'.format(name, queue_names[0]))
-    return [get_queue(name, autocommit=autocommit, queue_class=queue_class) for name in queue_names]
+        # or one queue with specified name
+        return [get_queue(*queue_names, **kwargs)]
+
+    # will return more than one queue
+    # import job class only once for all queues
+    kwargs['job_class'] = get_job_class(kwargs.pop('job_class', None))
+
+    queue_params = QUEUES[queue_names[0]]
+    connection_params = filter_connection_params(queue_params)
+    ret = [get_queue(queue_names[0], **kwargs)]
+
+    # do consistency checks while building return list
+    for name in queue_names[1:]:
+        queue = get_queue(name, **kwargs)
+        if queue.__class__ is not ret[0].__class__:
```

Comment:
I think we should use `isinstance` here.

Reply:
Well, that was the first thing I tried, but then `isinstance` matches all base classes. So if the first queue class is `DjangoRQ` and all others extend that, it will go through with no error.